npm/private/lifecycle/lifecycle-hooks.js

```diff
@@ -34,9 +34,29 @@ async function makeBins(nodeModulesPath, scope, segmentsUp) {
             'package.json'
         )
         if (fs.existsSync(packageJsonPath)) {
-            const packageJson = JSON.parse(
-                await fs.promises.readFile(packageJsonPath)
-            )
+            let packageJson = ''
+
+            // Catch invalid JSON errors and throw a more detailed Error.
+            // See bad quotes in puppeteer@8.0.0 for an example of bad json, or
```

Comment:
Were the bad qoutes just a red herring and the actual failure was the short read?

Reply:
I'm not 100% sure, so maybe I'll just drop this comment for now.